src/nisarqa/__main__.py

```diff
@@ -173,6 +173,71 @@ def load_user_runconfig(runconfig_yaml):
     return user_rncfg
 
 
+def generate_root_params(
+    runconfig_yaml: str, product_type: str
+) -> (
+    nisarqa.RSLCRootParamGroup
+    | nisarqa.GSLCRootParamGroup
+    | nisarqa.GCOVRootParamGroup
+    | nisarqa.RIFGRootParamGroup
+    | nisarqa.RUNWRootParamGroup
+    | nisarqa.GUNWRootParamGroup
+    | nisarqa.ROFFRootParamGroup
+    | nisarqa.GOFFRootParamGroup
+):
+    """
+    Generate a *RootParamGroup for `product_type` from a QA Runconfig yaml file.
+
+    The input runconfig file must follow the standard QA runconfig
+    format for `product_type`.
+    For an example runconfig template with default parameters (where available),
```

Comment:
The "where available" part was a little bit hard for me to parse when we looked over the code together. I think it's actually unnecessary for understanding this and maybe it'd be best to remove it.

```suggestion
    For an example runconfig template with default parameters,
```